lib/ansible/executor/task_executor.py

```diff
@@ -911,27 +911,13 @@ def _get_connection(self, cvars, templar):
             display.vvvv('attempting to start connection', host=self._play_context.remote_addr)
             display.vvvv('using connection plugin %s' % connection.transport, host=self._play_context.remote_addr)
 
-            options = self._get_persistent_connection_options(connection, cvars, templar)
+            options = connection.get_options(cvars)
```

Comment:
to get properly templated options, this should move to  L#541 after calling self._set_connection_options (or move this code block inside that function)

Reply:
also add sub_plugin handling to _set_connection_options